compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -2711,8 +2713,12 @@ impl<'a: 'ast, 'ast, 'tcx> LateResolutionVisitor<'a, '_, 'ast, 'tcx> {
         self.suggest_introducing_lifetime(
             &mut err,
             Some(lifetime_ref.ident.name.as_str()),
-            |err, _, span, message, suggestion| {
-                err.span_suggestion(span, message, suggestion, Applicability::MaybeIncorrect);
+            |err, _, span, message, suggestion, span_suggs| {
+                err.multipart_suggestion_verbose(
+                    message,
+                    std::iter::once((span, suggestion)).chain(span_suggs.iter().cloned()).collect(),
+                    Applicability::MaybeIncorrect,
+                );
```

Comment:
Since `span_suggestion` was changed to `multipart_suggestion_verbose`, this has lead to output changing in a lot of other scenarios that aren't strictly related to the inner/outer binder change as can be seen by all the UI tests you had to `--bless`.

Using the *verbose* style unconditionally isn't bad per se. However, I wonder if we could keep the non-verbose style if possible?

What happens if you just use `multipart_suggestion`? If the output isn't good in the inner/outer binder case, we could think about making the suggestion only verbose if we suggest removing inner binders:

```rs
let style = if span_suggs.is_empty() { SuggestionStyle::ShowCode } else { SuggestionStyle::ShowAlways };
err.multipart_suggestion_with_style(/*...*/, style);
```

Reply:
Done. Thank you.